src/main/java/chess/domain/piece/Position.java

```diff
@@ -0,0 +1,107 @@
+package chess.domain.piece;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Position {
+    public static final int MINIMUM_POSITION = 1;
+    public static final int MAXIMUM_POSITION = 8;
+
+    private final int x;
+    private final int y;
+
+    public Position(int x, int y) {
+        validateRange(x, y);
+        this.x = x;
+        this.y = y;
+    }
+
+    private void validateRange(int x, int y) {
+        validateColumn(x);
+        validateRow(y);
+    }
+
+    private void validateColumn(int x) {
+        if (x < MINIMUM_POSITION || x > MAXIMUM_POSITION) {
+            throw new IllegalArgumentException("올바르지 않은 열입니다.");
+        }
+    }
+
+    private void validateRow(int y) {
+        if (y < MINIMUM_POSITION || y > MAXIMUM_POSITION) {
+            throw new IllegalArgumentException("올바르지 않은 행입니다.");
+        }
+    }
+
+    public Set<Position> findMovablePositions(Set<Direction> directions) {
+        return directions.stream()
+                .filter(this::isInRange)
+                .map(direction -> new Position(direction.getDx() + x, direction.getDy() + y))
+                .collect(Collectors.toSet());
+    }
+
+    private boolean isInRange(Direction direction) {
+        int newX = direction.getDx() + x;
+        int newY = direction.getDy() + y;
+        return isLocationInRange(newX) && isLocationInRange(newY);
+    }
+
+    private boolean isLocationInRange(int location) {
+        return location >= MINIMUM_POSITION && location <= MAXIMUM_POSITION;
+    }
+
+    public Direction findDirectionTo(Position destination) {
+        int dx = destination.x - this.x;
+        int dy = destination.y - this.y;
+
+        if (isInEightDirection(dx, dy)) {
+            dx = calculateMoved(dx);
+            dy = calculateMoved(dy);
+            return Direction.findDirection(dx, dy);
+        }
+        throw new IllegalArgumentException("이동할 수 없습니다.");
+    }
```

Comment:
```suggestion
    public Direction findDirectionTo(Position destination) {
        int dx = destination.x - this.x;
        int dy = destination.y - this.y;

        if (isNotInEightDirection(dx, dy)) {
            throw new IllegalArgumentException("이동할 수 없습니다.");
        }
        
        dx = calculateMoved(dx);
        dy = calculateMoved(dy);
        return Direction.findDirection(dx, dy);
    }
```
예외상황이 발생할 때 터트리는 구조로 변경해보는 것도 괜찮을 것 같아요. 이런 방식을 사용하면 나중에 더 많은 예외상황이 생겼을 때 그 로직을 추가하기 편하거든요.

Reply:
예외상황이 발생할 때 먼저 터트리는 구조로 변경하겠습니다! 👍👍